frontend/src/Components/RatingSwitch.tsx

```diff
@@ -89,33 +89,39 @@ export default function RatingSwitch(props: RatingSwitchProps) {
     }, [showPopup, ratingRef, popupRef, votes, state.vote, props.id, props.type]);
 
     const handleVote = (vote: number) => {
-        const prevState = { ...state };
-        if (state.vote === vote) {
-            vote = 0;
-        }
-
-        const newRating = state.rating - (state.vote || 0) + vote;
-
-        setState({ vote: vote, rating: newRating});
-        if (props.onVote) {
-            props.onVote(newRating, vote);
-        }
-
-        api.voteAPI.vote(props.type, props.id, vote)
-            .then(result => {
-                setState({
-                    rating: result.rating,
-                    vote: result.vote
+        return (ev: React.MouseEvent) => {
```

Comment:
nit: could be simplified as  
`(vote: number) => (ev: React.MouseEvent) => { ... }`

Reply:
too hard to read to my taste

Reply:
Dunno what you are talking about, looks like idiomatic Haskell ;p